<launch>
    <arg name="slave_1_name" value="PSM2"/>
    <arg name="slave_2_name" value="PSM1"/>
    <arg name="master_1_name" value="MTMR"/>
    <arg name="master_2_name" value="MTML"/>
    <arg name="camera_namespace" value="camera"/>
    <arg name= "left_cam_name" value= "flea14150439" />
    <arg name= "right_cam_name" value= "flea14150441" />

    <!-- parameters used by some nodes regarding the transformations among the cams, arms and task-space -->
    <group ns="calibrations">
        <rosparam command="load" file="$(find teleop_vision)/launch/params_overlay.yaml" />
    </group>

    <!-- Node publishing the camera images -->
    <node pkg="teleop_vision" type="camera_capture_flea3" name="camera" output="screen"> </node>


    <!-- After some re-structuring of the architecture the dvrk_active_constraint_geomerty_generation
        node is not doing much right now and it will be soon removed.

    <node pkg="active_constraints" type="dvrk_active_constraint_geomerty_generation" name="dvrk_ac_geom" output="screen">
        <param name="number_of_arms" value="1"/>
        <param name="slave_1_name" value="$(arg slave_1_name)"/>
        <param name="slave_2_name" value="$(arg slave_2_name)"/>
        <param name="master_1_name" value="$(arg master_1_name)"/>
        <param name="master_2_name" value="$(arg master_2_name)"/>
        <param name="node_frequency" value="100"/>
    </node>
    -->

    <!-- The dvrk_active_constraint_enforcement node generates force based on the received desired pose and
        current pose of each master/slave couple.
    -->
    <node pkg="active_constraints" type="dvrk_active_constraint_enforcement" name="dvrk_ac_enforce" output="screen">
        <param name="slave_1_name" value="$(arg slave_1_name)"/>
        <param name="slave_2_name" value="$(arg slave_2_name)"/>
        <param name="master_1_name" value="$(arg master_1_name)"/>
        <param name="master_2_name" value="$(arg master_2_name)"/>
        <param name="node_frequency" value="100"/>
    </node>


</launch>
